LivekitUnitySampleApp/Assets/LivekitSamples.cs

```diff
@@ -309,7 +311,45 @@ public IEnumerator publishVideo()
     public void publishData()
     {
         var str = "hello from unity!";
-        room.LocalParticipant.PublishData(System.Text.Encoding.Default.GetBytes(str));
+
+        // Option 1: Using data streams
+        StartCoroutine(PerformSendText(str));
+
+        // Option 2: Using publish data
+        // PerformPublishData(str);
+    }
+
+    private IEnumerator PerformSendText(string message)
+    {
+        var sendTextCall = room.LocalParticipant.SendText(message, "my-topic");
+        yield return sendTextCall;
+
+        if (sendTextCall.IsError)
+        {
+            Debug.LogError("Failed to send text: " + sendTextCall.Error);
+            yield break;
+        }
+        Debug.Log("Text sent successfully");
+    }
+
+    private void PerformPublishData(string message)
+    {
+        var enc = System.Text.Encoding.Default.GetBytes(message);
+        room.LocalParticipant.PublishData(enc);
+    }
+
+    private IEnumerator HandleTextStream(TextStreamReader reader, string identity)
```

Comment:
The following code seems to have been removed in the last commit and forgotten to be added before room.connect?
```C#
	room.RegisterTextStreamHandler("my-topic", (data, identity) =>
                StartCoroutine(HandleTextStream(data, identity))
            );
```